docs/supervisor/development.md

```diff
@@ -21,6 +21,73 @@ The instructions here is for development of the Supervisor itself.
 Start the task "Run Supervisor" in Visual Studio Code, this will start an instance of the Supervisor inside the devcontainer you can use to test your changes.
 When the initializing is complete you can access the Home Assistant frontend on `http://localhost:9123`
 
+
+If you need to rebuild the base of the Supervisor run the task "Build Supervisor" and then restart the "Run Supervisor" task again when that completes.
+
+#### Using Windows WSL2
+
+While the system is Linux and compatible with Mac, it is also possible to build and test locally on Windows using WSL2 with Debian and other similar distros. This is achieved via Windows Subsystem for Linux 2 (WSL2), which makes POSIX tasks possible within Windows via system-level integration with a Linux virtual machine.
+
+:::tip
+You must use WSL2. WSL version 1 is not capable of running Docker and therefore is incompatible. If using this method, you must ensure you have [set up WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
+:::
+
+It is assumed at this point you have installed Debian from the Microsoft Store and [Visual Studio Code](https://code.visualstudio.com/download). From within your WSL2-enabled Debian terminal, run the following annotated bash script. If you copy the entire script it should perform the required setup of Debian for Home Assistant development but any problems may be difficult to troubleshoot, so it is recommended to run this line-by-line.
+
+```bash
+apt update; apt install -y curl wget git gnupg; #Update and install our dependencies
+curl https://raw.githubusercontent.com/creationix/nvm/master/install.sh | bash; #Install NVM
+source ~/.profile; nvm install node; #reload profile and install node
+sudo apt remove cmdtest yarn  #Remove problematic packages if installed. Debian current Yarn does not work.
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -  #Add yarn repo certificate
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list #Add Yarn repo
+sudo apt update; sudo apt install yarn -y; #Install yarn
+curl -fsSL https://get.docker.com |sudo bash  #install docker Community Edition
```

Comment:
This looks weird. Why is docker installed inside wsl. Any docker instance on windows is fully accessable from inside wsl instances.

It should be installed the official way in windows. Can you explain why that does not work?

Ps. Never done any supervisor development, so i might miss something.

Reply:
So that you can use the command line tools.

Reply:
Docker works inside wsl2 by default. Standard desktop install will inject the cli commands into your wsl2 images

Reply:
This is a new feature. Will verify if this is the case with Rancher Desktop.  I shouldn't recommend installing Docker Desktop.  https://www.suse.com/c/rancher_blog/using-rancher-desktop-for-local-kubernetes-development/#:~:text=Unlike%20Docker%E2%80%99s%20product%2C%20however%2C%20Rancher%20Desktop%20is%20a%20free%20and%20open%20source%20solution%20developed%20by%20SUSE

Reply:
Sure enough.  Good catch!  I created a new instance of Ubuntu Preview and enabled in  Settings->WSL->Integrations.  Docker was running immediately.  I can work with this.

![image](https://github.com/home-assistant/developers.home-assistant/assets/527919/a30ab97c-4f49-4646-ac9d-af3b90e4cd65)

Will get the system operational with this new guidance.  But probably this weekend.